Syndi Bundle Rework

AzuleUtama opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do

Updates most Syndicate bundles, as well as adding 3 new ones.

I've made a forum post already with the planned changes and the design document, which can be found here:

https://www.paradisestation.org/forum/topic/23098-syndicate-bundles-rebalancing-and-new-additions/

Some changes have been made based on feedback from the thread and discord, but the design document reflects all the changes present in this PR. To summarize the changes made to bundles here as well:

- **Spy** - Removed Garrote, Camera bug and AI detector. Added Storage Implanter and Energy Dagger.
- **Hacker** - Removed Syndicate AI board. Added Blue energy sword, Agent ID and EMP Kit.
- **Bond** - Removed 10mm HP magazine. Added 10mm AP magazine and Chameleon Sec HUD.
- **Thief** - Removed No-slips.
- **Payday** - Unchanged.
- **Implant** - Unchanged.
- **Darklord** - Added Power gloves and T-ray goggles.
- **Professional** - Removed Penetrator sniper rifle, .50 Soporific magazine and Energy dagger. Added Syndicate sniper rifle, .50 magazine and .50 penetrator magazine.
- **Sabotage** - Removed.

The new bundles are:

- **Agent 13** - Contains Stechkin, Silencer, 10mm magazine, 10mm HP magazine, Fiber wire, Access Tuner, Chameleon thermal glasses, Freedom implanter, Chameleon jumpsuit, Agent ID, Gold coin and Syndicate encryption key.
- **Infiltrator** - Contains Experimental teleporter, Experimental krav maga gloves, Chameleon thermal glasses, Advanced pinpointer, Boxed space suit, Preloaded RCD, Single-use autoimplanter loaded with meson eyes and Syndicate encryption key.
- **Grenadier** - Contains Tactical grenadier belt (Contains 2 Frag and EMP grenades, 5 C4 Explosives, 5 Smoke and Gluon grenades and 1 Minibomb grenade), Stechkin, 2x 10mm fire magazine, No-slip shoes, Blood-red hardsuit, Combat gloves, Agent ID and Syndicate encryption key.

Four additional changes made are:

- Syndicate bundles are now a subtype of syndie_kit, rather than the standard box. This means they now spawn in the red and black syndie boxes and don't look out of place in the uplink_kits.dm file with everything else.
- Penetrator Sniper Rifle removed, the kit using it now uses a regular sniper so this isn't needed anymore.
- New single-use autosurgeon added containing meson eyes for the Infiltrator bundle. Code for single-use autosurgeons already exists but is currently unused for the most part.
- New variant of grenadier belt added containing fewer slots and a different grenade layout for the Grenadier bundle.

<!-- Include a small to medium description of what your PR changes. Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Bundles haven't really been touched since 2020 and in that time there's been several changes to syndi items, combat as well as TC costs. I also noticed certain bundles were generally disliked by players and have majorly reworked these ones in particular. Adding a few new ones also helps keep the item interesting to anyone willing to part with all their TC.

<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif or mp4 of your feature if you want. -->

## Testing

Spawned several kits in and made sure they contained everything they should, as well as making sure the unique items worked properly.

<!-- How did you test the PR, if at all? -->

## Changelog
:cl:
tweak: Syndicate bundles reworked; 3 new ones added, 1 removed and 6 have been updated.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

AzuleUtama commented:
> How does nerfing most of the bundles encourage people to try them out, exactly?

Could you clarify which ones in particular were nerfed? From my perspective, only two bundles were directly nerfed and I wouldn't argue they were major ones:

Implanter had the SEK removed since it's not an implant and the bundle comes with almost every syndicate bio-chip with 10TC refunded back - it's by far one of the best ones you can get and doesn't need the free SEK at all.

Thief had the SEK and No-slips removed for reasons stated in the design document - it has a very high TC value already with some powerful items included (notably the energy crossbow, chameleon projector and pickpocket gloves + agent ID combo) and really didn't need these on top of all that. They also don't really fit the theme that well, with no-slips no longer muffling footsteps like they used which is the reason I added them in a few years ago.

KittyNoodle commented:
my worry with the dark lord thing is that (from experience doing many gimmick antags) people will see wizard spells and valid you immediately, then get confused when the round doesn't end when the 'wizard' died

Carthusia commented:
Some of the changes feel, odd?

Like removeing the spys camera bug and AI detector, two items that are invaluable for spying and keeping security from noticing you as you can see where people are, and make sure the AI isnt watching you, and replacing them with an Edagger, a weapon that is so un-stealthy that its kind of a pointless weapon when it comes to being a spy.

Its just as loud as a normal Esword, easily spottable when in use and offers no boon beyond a deadly weapon that will get you perma brigged if ever caught with it, nullifying the idea of deniability.

Personally id say swap the Edagger out and return the camera bug and AI detector, both combined are 2 TC so there is no point rebalance required. I do agree with the garrote removal, that never fit the idea of a spy bundle.

Hacker seems like its no longer a hacker bundle, it includes nothing that assists with hacking except a joke item that gives the AI random ion laws, but in no way directly assists the agent in question, a blue Esword instead of somthing that lets you listen to the thing you would be using the kit to hack and no board to hack it with.

Im not entirely sure why the bond kit lost its suppressor, its kind of a bond trademark to use a suppressor and its EMP kit loss removes the idea of a gadget based kit, which is another hallmark of bond.

Another thing that seems kinda weird to me is the removal of so many of the kits syndicate com key, a key item that allows bundle users to contact eachother and group up with their gimmicks.

Even somthing as minor as a voice changer for the dark lord is gone, and the whole point is to mask your voice like a sith lord with a spooky mask, of which is now gone.

Dafrek commented:
Most of these are pretty neat changes, although giving a traitor the force wall spell might be a bit too strong.

McRamon commented:
I love it but just noticed that removal of agent id/voice changer goes a bit against the ability to give yourself proper sith persona. Also force Wall is indeed very strong. I'd probably swap wall back with id and vc

Coolrune206 commented:
I believe the SEK was in all the Bundles to encourage traitors to use it to communicate. Losing SEK from a bunch of bundles seems odd as a result.

Edan52 commented:
Will have a deeper look at the kits in a bit, looks mostly fine but I will never approve traitors getting forcewall. Going to put an objection in for that reason alone but if that's changed I'll remove it.

matttheficus commented:
- SEK should stay in every kit. The idea is that Bundles are more built for newer players who arent sure what to buy in my opinion.
- Darklord should not have spells, period. Instant Summons has some nasty interactions with lockers (you can teleport lockers full of gear with it and its non-removable) and force wall is too strong.y
- Hacker should retain its AI Law Board and Binary Key, its a hacker kit after all.
- Bond should keep the silencer.
- The Sniper kit is still mega-ultra-super-garbage, it really should just be removed.

AzuleUtama commented:
In response to some of the feedback already given:

**Darklord**
Will remove the forcewall spellbook and add the voice changer, agent ID and SEK back in. I thought that forcewall would work well with TK, but underestimated how strong this spell would be for traitors and the issues it could cause - not to mention it's gotten an objection. I would rather Instant Summons still remains since TK alone doesn't really make this an especially good kit with everything else it contains.

**Syndi Encryption Keys**
Owing to a lot of feedback on these I'll add them back into all bundles rather than just specific ones. I did originally feel that highlighted codewords kind of made these redundant to some degree, but they are still far superior for traitors working in a team and my other argument that not all traitors would want them is a flawed reason for mass removing them on re-examination, since they can always just choose not to use them.

**Hacker**
I'm not keen on reverting this back to being a "Hack the AI" bundle, as I raised several points in the design document on why theming the bundle around this specifically isn't a great concept:

- It’s a bit rough on newbies since you need to build/access an AI uploading console.
- If the AI is Malf, it will likely ignore you.
- If the AI is nonexistent, dead, AFK or just bald it becomes useless.
- If multiple traitors have this bundle or the hacked board it can also become useless if you can work with them.
- Some AIs might try to work around your hacked law out of spite if it isn’t worded well.

I'd rather this stuck to being a kit themed around Opposing the AI instead, though I will admit that it's a bit weaker than I may have intended and could be improved by re-adding the binary key should you get creative with ion laws or get a freeform board as well as an Agent ID due to it's somewhat hidden but fitting features of making beepsky ignore your crimes and the AI unable to track you.

**Bond**
Minor point raised by some, but adding the Silencer and EMP kit back in would probably be a good idea looking at this again. I wanted to have bond differ from the new Agent bundle a bit more by removing it but getting a silencer in maintenance or even R&D is a possibility anyway, so this isn't a great change and they differ in ammo types anyway. It also lacks a lot of utility and doesn't need to lose the EMP kit even if the kit does contain CQC.

Odieman1 commented:
Some thoughts, I think you have overall good ideas:

- But...  I feel the Agent box and Bond box are too similar, or at least fulfill similar roles.  They are both stealthy bundles with some utility gadgets and suppressed stechkins.  Agent 007 vs Agent 13?  pamtheyrethesamepicture.jpg

- I agree with sabotage being removed, though I think it might be better to replace it with a possible 'Hijacker' box you can only get as Hijacker.  Something that comes with Hijack only tools and then some.  Many ideas can be decided here.

- The infiltrator is just cool.  I like that box.  It feels in-line with everything that exists already.  Give it a SEK though.

- I don't like the Professional box.  If you do not have a kill objective, the utilities in this box are practically non-existent.  Sniper rifles as a weapon are loud, large, and impossible to not be noticed with.  An entire kit built around a sniper rifle sounds cool, but it really isn't on a cramped space station.  Feel free to tell me I'm wrong here, I welcome conversation about it.

When looking at these bundles, there is a general idea that each have, generally balancing between offensive <------> utility but you could argue there is a compass of ideas instead.  For Operative you are probably very offensive so...  I agree with your options.  But the options themselves present problems from the perspective of the crew.  Maybe instead of a blood red hardsuit, they get a bulletproof vest and a red headband.  Maybe a combat knife too.  Something that more or less cements them as an offensive soldier of fortune that doesn't so much invite valid hunting and cargo SMG's upon their arrival.

_From here on out, I'm only going to speculate on my own ideas._

A support box!  All these boxes came with teamwork in mind.  Make a box with teamwork promised.  Medibeam gun, Surgery arm auto-implant doc, Compact Defib...  But also some offensive tools inline with a rogue doc.  Rapid Syringe gun, Viral Injector, Combat Stimms.  These are just ideas.

Half of the bundles are pop-culture references.  In your design philosophy, I feel like this is somewhat important but not required.

Overall, keep discussing ideas and iterating.  I don't think we should cement anything yet until a lot of people agree to it.  There will always be contention, but listen to the debates presented.  Hope this helps in some way!

Xsingular commented:
I am in love with that Agent 13 kit, should come with an obvious barcode that people still somehow completely miss.

AzuleUtama commented:
Made following changes based on feedback:

- All bundles now have SEKs again, including the new ones.
- Removed spellbooks from Darklord and added original items back in. Now includes Power Gloves and T-ray optical glasses as these fit the theme and can be countered/disarmed unlike the spells originally suggested.
- Adds the EMP Kit and Silencer back to Bond.
- Adds Binary key back to Hacker and additionally adds Agent ID card. TC Value is now the same as the old kit.
- Removes microbomb implanter and replaces with X4 explosive in Operative. Given this kit will likely cause some chaos due to the nuke ops similarities there should be some kind of tell that they weren't an actual operative - this being them not exploding on death.

Will update the original PR information to reflect the new changes as well.

AzuleUtama commented:
Following more feedback regarding operative being too strong/problematic, have reworked the bundle into "grenadier" having a high focus on pyrotechnics - both the damaging and more utility kind.

Grenadier contains the following:
- Tactical grenadier belt (Watered down version of the nukie grenade belt with a stronger focus on utility grenades and explosives. Contains 2 Frag and EMP grenades, 5 C4 Explosives, 5 Smoke and Gluon grenades and 1 Minibomb grenade)
- Stechkin
- 2x 10mm Fire magazine
- Blood-red hardsuit
- Combat Gloves
- Agent ID
- No-slips shoes
- Syndi encryption key

As with the last change, will update the original post reflecting all changes.

S34NW commented on code/game/objects/items/weapons/storage/uplink_kits.dm:
```diff
@@ -1,124 +1,148 @@
 		/obj/item/encryptionkey/syndicate) // 2TC
 
-	var/static/list/payday = list( // 36TC + four 0TC
+	var/static/list/payday = list( // 35TC
```
```suggestion
	var/static/list/payday = list( // 36TC
```
This is still 36 TC if you add up the stated values

AzuleUtama replied:
It's 35TC, I just forgot to update the jammer to be 4TC in the comments so I corrected that.

S34NW reviewed (changes requested):
one minor change
